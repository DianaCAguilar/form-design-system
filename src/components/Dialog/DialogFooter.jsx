import React, { useContext } from 'react';
import cc from 'classcat';
import DialogContext from './DialogContext';

const DialogFooter = () => {
  const { footerContent, alwaysShowBorder, isOverflowing } = useContext(DialogContext);
  return (
    <>
      {footerContent && (
        <div className="dialog-footer">
          <div
            className={cc([
              {
                'border--top': alwaysShowBorder || isOverflowing,
              },
              'bgColor--white',
              'padding--all',
            ])}
          >
            {footerContent}
          </div>
        </div>
      )}
    </>
  );
};

export default DialogFooter;
