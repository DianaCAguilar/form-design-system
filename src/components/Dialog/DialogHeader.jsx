import React, { useContext } from 'react';
import IconButton from 'components/IconButton';
import DenyIcon from 'lib/icons/react/DenyIcon';
import DialogContext from './DialogContext';

const DialogHeader = () => {
  const { onDismiss, title } = useContext(DialogContext);
  return (
    <>
      {(title || onDismiss) && (
        <div className="bgColor--white border--bottom">
          <div className="dialog-header padding--left padding--top padding--bottom">
            <div className="type--head4">{title}</div>
            {onDismiss && (
              <div className="dialog-icon">
                <IconButton
                  Icon={DenyIcon}
                  onClick={onDismiss}
                  aria-label="Close"
                  label="Close"
                />
              </div>
            )}
          </div>
        </div>
      )}
    </>
  );
};

export default DialogHeader;
