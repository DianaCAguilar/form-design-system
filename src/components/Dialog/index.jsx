import React, { useEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import rafSchd from 'raf-schd';
import DenyIcon from 'lib/icons/react/DenyIcon';
import IconButton from 'components/IconButton';
import { DialogOverlay, DialogContent } from '@reach/dialog';
import { AnimatePresence, motion } from 'framer-motion';
import cc from 'classcat';

export const isElementOverflowing = ({ current }) => {
  // Checking for current first is safer just in case,
  // and will also fail gracefully in tests
  if (current) {
    return current.scrollHeight > current.clientHeight;
  }
  return false;
};

const Dialog = ({
  width = '500px',
  height = '80vh',
  disableFocusTrap = false,
  onDismiss,
  alwaysShowBorder,
  isOpen,
  title,
  content,
  footerContent,
}) => {
  const Core = () => {
    const [isOverflowing, setIsOverflowing] = useState(false);

    const contentEl = useRef(null);

    const handleResize = () => {
      rafSchd(setIsOverflowing(isElementOverflowing(contentEl)));
    };

    useEffect(() => {
      if (!alwaysShowBorder) {
        handleResize(); // needs to fire once immediately on mount
        // eslint-disable-next-line no-undef
        window.addEventListener('resize', handleResize);
        return () => {
          // eslint-disable-next-line no-undef
          window.removeEventListener('resize', handleResize);
        };
      }
      return undefined;
    }, [alwaysShowBorder]);

    return (
      <>
        {(title || onDismiss) && (
          /* Do not mess with / combine the 2 lines of following classes unless you
           *thoroughly* vet responsive, header props, and long text, all at once */
          <div className="bgColor--white border--bottom">
            <div className="dialog-header padding--left padding--top padding--bottom">
              {title && (
                <span id="a11y-dialog-title" className="type--head4">
                  {title}
                </span>
              )}
              {onDismiss && (
                <div className="dialog-icon">
                  <IconButton
                    Icon={DenyIcon}
                    onClick={onDismiss}
                    aria-label="Close"
                    label="Close"
                  />
                </div>
              )}
            </div>
          </div>
        )}
        <div className="dialog-content" ref={contentEl}>
          <div className="padding--all bgColor--white">{content}</div>
        </div>
        {footerContent && (
          <div className="dialog-footer">
            <div
              className={cc([
                {
                  'border--top': alwaysShowBorder || isOverflowing,
                },
                'bgColor--white',
                'padding--all',
              ])}
            >
              {footerContent}
            </div>
          </div>
        )}
      </>
    );
  };

  const MotionDialogOverlay = motion(DialogOverlay);
  const MotionDialogContent = motion(DialogContent);

  return (
    <AnimatePresence>
      {isOpen && (
        <MotionDialogOverlay
          initial={{ opacity: 0 }}
          animate={{ opacity: 1 }}
          exit={{ opacity: 0 }}
          onDismiss={onDismiss}
          dangerouslyBypassFocusLock={disableFocusTrap}
        >
          <MotionDialogContent
            aria-label={title}
            className="elevation--3"
            style={{
              maxWidth: width,
              maxHeight: height,
            }}
          >
            <Core />
          </MotionDialogContent>
        </MotionDialogOverlay>
      )}
    </AnimatePresence>
  );
};

Dialog.propTypes = {
  /** Controls whether the modal (and overlay) are shown or not */
  isOpen: PropTypes.bool,

  /** Add into bottom portion of modal */
  footerContent: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),

  /** Add into content (middle) portion of modal */
  content: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,

  /** Controls text displayed in modal header */
  title: PropTypes.string,

  /**
   * Callback that user can pass in, to be conditionally fired when
   * user attempts to close modal. When defined, the modal close button appears
   */
  onDismiss: PropTypes.func,

  /** Custom modal width */
  width: PropTypes.string,

  /** Custom modal height */
  height: PropTypes.string,

  /**
   * Border between content area and footer currently renders only if there is scrollable content. Set this boolean
   * so that the border always renders no matter what.
   */
  alwaysShowBorder: PropTypes.bool,

  /**
   * Disables the focus trap on the Dialog.
   * Useful when the Dialog contains components that manage focus (e.g. `Menu`)
   */
  disableFocusTrap: PropTypes.bool,
};

export default Dialog;
