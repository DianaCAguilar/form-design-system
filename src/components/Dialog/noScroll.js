// Adapted from the following:
// https://github.com/davidtheclark/no-scroll/blob/master/index.js
/* eslint-disable */

(function(root) {
  let isOn = false;
  let scrollbarSize;
  let scrollTop;

  function getScrollbarSize() {
    if (typeof scrollbarSize !== 'undefined') return scrollbarSize;

    const doc = document.documentElement;
    const dummyScroller = document.createElement('div');
    dummyScroller.setAttribute(
      'style',
      'width:99px;height:99px;' + 'position:absolute;top:-9999px;overflow:scroll;'
    );
    doc.appendChild(dummyScroller);
    scrollbarSize = dummyScroller.offsetWidth - dummyScroller.clientWidth;
    doc.removeChild(dummyScroller);
    return scrollbarSize;
  }

  function hasScrollbar() {
    return document.documentElement.scrollHeight > window.innerHeight;
  }

  function on(options) {
    if (typeof document === 'undefined' || isOn) return;
    const doc = document.documentElement;
    if (hasScrollbar()) {
      doc.style.width = `calc(100% - ${getScrollbarSize()}px)`;
    } else {
      doc.style.width = '100%';
    }
    doc.style.overflow = 'hidden';
    isOn = true;
  }

  function off() {
    if (typeof document === 'undefined' || !isOn) return;
    const doc = document.documentElement;
    doc.style.width = '';
    doc.style.overflow = '';
    isOn = false;
  }

  function toggle() {
    if (isOn) {
      off();
      return;
    }
    on();
  }

  const noScroll = {
    on,
    off,
    toggle,
  };

  if (typeof module !== 'undefined' && typeof module.exports !== 'undefined') {
    module.exports = noScroll;
  } else {
    root.noScroll = noScroll;
  }
})(this);
