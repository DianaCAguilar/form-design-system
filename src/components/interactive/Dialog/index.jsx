import React from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';
import { CSSTransition } from 'react-transition-group';
import DenyIcon from '../../../../lib/icons/react/DenyIcon';

const onKeyDown = (e) => e.keyCode === 27;

const Dialog = (props) =>
  ReactDOM.createPortal(
    <CSSTransition timeout={200}>
      <React.Fragment>
        {props.isOpen && (
          <React.Fragment>
            <div className="dialog-overlay" onClick={props.onDismiss}></div>
            <div
              className="dialog elevation--3"
              role="dialog"
              tabIndex="-1"
              onKeyDown={(e) => onKeyDown(e) && props.onDismiss()}
            >
              {(props.title || props.canDismiss) && (
                <React.Fragment>
                  <div className="dialog-header">
                    <div className="dialog-title fontSize--xl">{props.title}</div>
                    {props.canDismiss && (
                      <button className="dialog-icon" onClick={props.onDismiss}>
                        <DenyIcon size="xs" />
                      </button>
                    )}
                  </div>
                  <div className="dialog-divider" />
                </React.Fragment>
              )}
              <div className="dialog-content">{props.content}</div>
              {props.footerContent && (
                <div className="dialog-footer">{props.footerContent}</div>
              )}
            </div>
          </React.Fragment>
        )}
      </React.Fragment>
    </CSSTransition>,
    // eslint-disable-next-line no-undef
    document.body
  );

Dialog.propTypes = {
  isOpen: PropTypes.bool,
  canDismiss: PropTypes.bool,
  footerContent: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  content: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,
  title: PropTypes.string,
};

export default Dialog;
