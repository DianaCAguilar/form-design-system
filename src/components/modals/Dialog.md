An accessible base modal for all other modals to be built upon (either your own, or other FDS modals like `<Prompt />`)