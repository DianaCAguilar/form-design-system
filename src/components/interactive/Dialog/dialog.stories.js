import React from 'react';

import { storiesOf } from '@storybook/react';
import { withKnobs, text, boolean } from '@storybook/addon-knobs';

import Dialog from '.';

storiesOf('Interactive/Dialog', module)
  .addDecorator(withKnobs)
  .add('Knobs', () => (
    <Dialog
      isOpen={boolean('isOpen', true)}
      footerContent={text('footerContent', '')}
      content={text('content', 'Modal content here')}
      canDismiss={boolean('canDismiss', true)}
      title={text('title', 'Modal title here')}
    />
  ));
