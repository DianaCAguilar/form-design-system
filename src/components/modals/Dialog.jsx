import React, { useEffect, useLayoutEffect, useRef, useState } from 'react';
import PropTypes from 'prop-types';
import ReactDOM from 'react-dom';
import { CSSTransition } from 'react-transition-group';
import FocusTrap from 'focus-trap-react';
import noScroll from 'no-scroll';
import rafSchd from 'raf-schd';
import DenyIcon from '../../../lib/icons/react/DenyIcon';
import Flex from '../layout/Flex';
import FlexItem from '../layout/FlexItem';
import Section from '../layout/Section';

export const isElementOverflowing = ({ current }) => {
  // Checking for current first is safer just in case,
  // and will also fail gracefully in tests
  if (current) {
    return current.scrollHeight > current.clientHeight;
  }
  return false;
};

const Dialog = (props) => {
  const contentEl = useRef(null);

  const handleKeyDown = (e) => {
    if (props.onDismiss && e.key === 'Escape') {
      props.onDismiss();
    }
  };

  const [isOverflowing, setIsOverflowing] = useState(false);

  const handleResize = () => {
    rafSchd(setIsOverflowing(isElementOverflowing(contentEl)));
  };

  useEffect(() => {
    handleResize(); // needs to fire once immediately on mount
    // eslint-disable-next-line no-undef
    window.addEventListener('resize', handleResize);
    noScroll.off();
    return () => {
      noScroll.off();
      // eslint-disable-next-line no-undef
      window.removeEventListener('resize', handleResize);
    };
  }, []);

  useLayoutEffect(() => {
    // This toggles scrolling on and off based on whether the modal
    // is shown or not
    if (props.isOpen) {
      noScroll.on();
    } else {
      noScroll.off();
    }
  }, [props.isOpen]);

  return ReactDOM.createPortal(
    <CSSTransition timeout={200} in={props.isOpen} classNames="dialog" unmountOnExit>
      <React.Fragment>
        <FocusTrap>
          <div>
            <div className="dialog-overlay" onClick={props.onDimiss}></div>
            <div className="dialog-wrapper">
              <div
                className="dialog elevation--3"
                role={props.role}
                aria-labelledby={props.title && 'a11y-dialog-title'}
                aria-describedby="a11y-dialog-desc"
                tabIndex="-1"
                aria-modal="true"
                onKeyDown={handleKeyDown}
              >
                {(props.title || props.onDismiss) && (
                  <React.Fragment>
                    <div className="dialog-header">
                      <Section border="bottom">
                        <Flex justify="spaceBetween" align="start">
                          <FlexItem>
                            {props.title && (
                              <div
                                className="dialog-title type--head4"
                                id="a11y-dialog-title"
                              >
                                {props.title}
                              </div>
                            )}
                          </FlexItem>
                          {props.onDismiss && (
                            <FlexItem shrink>
                              <button className="dialog-icon" onClick={props.onDismiss}>
                                <DenyIcon size="xs" />
                              </button>
                            </FlexItem>
                          )}
                        </Flex>
                      </Section>
                    </div>
                  </React.Fragment>
                )}
                <div className="dialog-content" ref={contentEl}>
                  <Section>{props.content}</Section>
                </div>
                {props.footerContent && (
                  <div className="dialog-footer">
                    <Section border={isOverflowing ? 'top' : undefined} bgColor="white">
                      {props.footerContent}
                    </Section>
                  </div>
                )}
              </div>
            </div>
          </div>
        </FocusTrap>
      </React.Fragment>
    </CSSTransition>,
    // eslint-disable-next-line no-undef
    document.body
  );
};

Dialog.defaultProps = {
  role: 'dialog',
};

Dialog.propTypes = {
  /** Controls the role of the modal */
  role: PropTypes.oneOf(['dialog', 'alertdialog']),
  /** Controls whether the modal (and overlay) are shown or not */
  isOpen: PropTypes.bool,
  /** Add into bottom portion of modal */

  footerContent: PropTypes.oneOfType([PropTypes.node, PropTypes.string]),
  /** Add into content (middle) portion of modal */

  content: PropTypes.oneOfType([PropTypes.node, PropTypes.string]).isRequired,
  /** Controls text displayed in modal header */

  title: PropTypes.string,
  /**
   * Callback that user can pass in, to be conditionally fired when
   * user attempts to close modal. Also controls modal x
   */
  onDismiss: PropTypes.func,
};

export default Dialog;
