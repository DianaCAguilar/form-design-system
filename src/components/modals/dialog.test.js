import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'
import Dialog from './Dialog'

describe('Dialog', () => {

  const dismiss = jest.fn();

  it('tests that dialog renders properly', () => {
    // Scrolling should not be locked before dialog is open
    expect(window.getComputedStyle(document.body).overflow).toBe('');

    render(
      <Dialog
        content={<button>content</button>}
        isOpen
        footerContent={<div>footerContent</div>}
        title="title"
        onDismiss={dismiss}
        height="1000px"
        width="1000px"
        alwaysShowBorder
      />
    )
    /* Test that content renders properly */
    /* ================================== */
    expect(screen.getByText('content')).toBeTruthy();
    expect(screen.getByText('footerContent')).toBeTruthy();
    expect(screen.getByText('title')).toBeTruthy();

    /* Tests that onDismiss gets fired correctly */
    /* ========================================= */
    // The close button is the first thing focused / active
    fireEvent.click(document.activeElement);
    fireEvent.click(screen.getByLabelText('Close'));
    // Pressing esc anywhere in dialog should trigger a close
    fireEvent.keyDown(screen.getByText('content'), { key: 'Escape', code: 27 })
    expect(dismiss).toHaveBeenCalledTimes(3)


    /* Tests styles */
    /* ============ */
    expect(window.getComputedStyle(screen.getByRole('dialog'))['max-height']).toBe('1000px');
    expect(window.getComputedStyle(screen.getByRole('dialog'))['max-width']).toBe('1000px');
    // Scrolling should be locked when dialog is open
    expect(window.getComputedStyle(document.body).overflow).toBe('hidden');
  })

})
