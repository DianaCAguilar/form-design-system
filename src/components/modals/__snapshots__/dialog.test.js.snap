// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog component matches snapshot (default props) 1`] = `
<Fragment>
  <Portal
    containerInfo={
      <body>
        <div>
          <div
            class="dialog-overlay"
          />
          <div
            class="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-modal="true"
              class="dialog elevation--3"
              role="dialog"
              style="max-width: 500px;"
              tabindex="-1"
            >
              <div
                class="dialog-header"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom border--bottom"
                >
                  <div
                    class="padding--right--double type--head4"
                  >
                     
                     
                  </div>
                  <div
                    class="dialog-icon"
                  >
                    <button
                      aria-label="close"
                      class="fdsIconButton rounded--all border--focus--noTransition transition--default fdsIconButton--ghost fdsIconButton--m"
                      role="img"
                    >
                      <span
                        class=""
                      >
                        <div
                          class="fds-icon fds-icon--s"
                          style="width: 18px; height: 18px;"
                        >
                          <svg
                            viewBox="0 0 24 24"
                          >
                            <g
                              fill-rule="evenodd"
                            >
                              <g>
                                <polygon
                                  points="10.363961 11.7781746 4 5.41421356 5.41421356 4 11.7781746 10.363961 18.1421356 4 19.5563492 5.41421356 13.1923882 11.7781746 19.5563492 18.1421356 18.1421356 19.5563492 11.7781746 13.1923882 5.41421356 19.5563492 4 18.1421356"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="dialog-content"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom"
                >
                  <button>
                    hey
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="dialog-overlay"
          />
          <div
            class="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-modal="true"
              class="dialog elevation--3"
              role="dialog"
              style="max-width: 500px;"
              tabindex="-1"
            >
              <div
                class="dialog-header"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom border--bottom"
                >
                  <div
                    class="padding--right--double type--head4"
                  >
                     
                     
                  </div>
                  <div
                    class="dialog-icon"
                  >
                    <button
                      aria-label="close"
                      class="fdsIconButton rounded--all border--focus--noTransition transition--default fdsIconButton--ghost fdsIconButton--m"
                      role="img"
                    >
                      <span
                        class=""
                      >
                        <div
                          class="fds-icon fds-icon--s"
                          style="width: 18px; height: 18px;"
                        >
                          <svg
                            viewBox="0 0 24 24"
                          >
                            <g
                              fill-rule="evenodd"
                            >
                              <g>
                                <polygon
                                  points="10.363961 11.7781746 4 5.41421356 5.41421356 4 11.7781746 10.363961 18.1421356 4 19.5563492 5.41421356 13.1923882 11.7781746 19.5563492 18.1421356 18.1421356 19.5563492 11.7781746 13.1923882 5.41421356 19.5563492 4 18.1421356"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="dialog-content"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom"
                >
                  <button>
                    hey
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="dialog-overlay"
          />
          <div
            class="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-modal="true"
              class="dialog elevation--3"
              role="dialog"
              style="max-width: 500px;"
              tabindex="-1"
            >
              <div
                class="dialog-content"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom"
                >
                  <button>
                    hey
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </body>
    }
  >
    <CSSTransition
      classNames="dialog"
      in={true}
      timeout={200}
      unmountOnExit={true}
    >
      <FocusTrap
        _createFocusTrap={[Function]}
        active={true}
        focusTrapOptions={Object {}}
        paused={false}
      >
        <div>
          <div
            className="dialog-overlay"
          />
          <div
            className="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-modal="true"
              className="dialog elevation--3"
              onKeyDown={[Function]}
              role="dialog"
              style={
                Object {
                  "maxWidth": "500px",
                }
              }
              tabIndex="-1"
            >
              <div
                className="dialog-content"
              >
                <Section
                  bgColor="white"
                  xPadding="default"
                  yPadding="default"
                >
                  foo
                </Section>
              </div>
            </div>
          </div>
        </div>
      </FocusTrap>
    </CSSTransition>
  </Portal>
</Fragment>
`;

exports[`Dialog component matches snapshot (set all props) 1`] = `
<Fragment>
  <Portal
    containerInfo={
      <body>
        <div>
          <div
            class="dialog-overlay"
          />
          <div
            class="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-modal="true"
              class="dialog elevation--3"
              role="dialog"
              style="max-width: 500px;"
              tabindex="-1"
            >
              <div
                class="dialog-header"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom border--bottom"
                >
                  <div
                    class="padding--right--double type--head4"
                  >
                     
                     
                  </div>
                  <div
                    class="dialog-icon"
                  >
                    <button
                      aria-label="close"
                      class="fdsIconButton rounded--all border--focus--noTransition transition--default fdsIconButton--ghost fdsIconButton--m"
                      role="img"
                    >
                      <span
                        class=""
                      >
                        <div
                          class="fds-icon fds-icon--s"
                          style="width: 18px; height: 18px;"
                        >
                          <svg
                            viewBox="0 0 24 24"
                          >
                            <g
                              fill-rule="evenodd"
                            >
                              <g>
                                <polygon
                                  points="10.363961 11.7781746 4 5.41421356 5.41421356 4 11.7781746 10.363961 18.1421356 4 19.5563492 5.41421356 13.1923882 11.7781746 19.5563492 18.1421356 18.1421356 19.5563492 11.7781746 13.1923882 5.41421356 19.5563492 4 18.1421356"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="dialog-content"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom"
                >
                  <button>
                    hey
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="dialog-overlay"
          />
          <div
            class="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-modal="true"
              class="dialog elevation--3"
              role="dialog"
              style="max-width: 500px;"
              tabindex="-1"
            >
              <div
                class="dialog-header"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom border--bottom"
                >
                  <div
                    class="padding--right--double type--head4"
                  >
                     
                     
                  </div>
                  <div
                    class="dialog-icon"
                  >
                    <button
                      aria-label="close"
                      class="fdsIconButton rounded--all border--focus--noTransition transition--default fdsIconButton--ghost fdsIconButton--m"
                      role="img"
                    >
                      <span
                        class=""
                      >
                        <div
                          class="fds-icon fds-icon--s"
                          style="width: 18px; height: 18px;"
                        >
                          <svg
                            viewBox="0 0 24 24"
                          >
                            <g
                              fill-rule="evenodd"
                            >
                              <g>
                                <polygon
                                  points="10.363961 11.7781746 4 5.41421356 5.41421356 4 11.7781746 10.363961 18.1421356 4 19.5563492 5.41421356 13.1923882 11.7781746 19.5563492 18.1421356 18.1421356 19.5563492 11.7781746 13.1923882 5.41421356 19.5563492 4 18.1421356"
                                />
                              </g>
                            </g>
                          </svg>
                        </div>
                      </span>
                    </button>
                  </div>
                </div>
              </div>
              <div
                class="dialog-content"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom"
                >
                  <button>
                    hey
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div>
          <div
            class="dialog-overlay"
          />
          <div
            class="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-modal="true"
              class="dialog elevation--3"
              role="dialog"
              style="max-width: 500px;"
              tabindex="-1"
            >
              <div
                class="dialog-content"
              >
                <div
                  class="bgColor--white display--block padding--left padding--right padding--top padding--bottom"
                >
                  <button>
                    hey
                  </button>
                </div>
              </div>
            </div>
          </div>
        </div>
      </body>
    }
  >
    <CSSTransition
      classNames="dialog"
      in={true}
      timeout={200}
      unmountOnExit={true}
    >
      <FocusTrap
        _createFocusTrap={[Function]}
        active={true}
        focusTrapOptions={Object {}}
        paused={false}
      >
        <div>
          <div
            className="dialog-overlay"
          />
          <div
            className="dialog-zIndex dialog-wrapper"
          >
            <div
              aria-describedby="a11y-dialog-desc"
              aria-labelledby="a11y-dialog-title"
              aria-modal="true"
              className="dialog elevation--3"
              onKeyDown={[Function]}
              role="alertdialog"
              style={
                Object {
                  "maxWidth": "500px",
                }
              }
              tabIndex="-1"
            >
              <div
                className="dialog-header"
              >
                <Section
                  bgColor="white"
                  border="bottom"
                  xPadding="default"
                  yPadding="default"
                >
                  <div
                    className="padding--right--double type--head4"
                  >
                    <span
                      id="a11y-dialog-title"
                    >
                      hey
                    </span>
                     
                  </div>
                  <div
                    className="dialog-icon"
                  >
                    <IconButton
                      Icon={[Function]}
                      aria-label="close"
                      onClick={[Function]}
                      radius="square"
                      size="m"
                      theme="ghost"
                    />
                  </div>
                </Section>
              </div>
              <div
                className="dialog-content"
              >
                <Section
                  bgColor="white"
                  xPadding="default"
                  yPadding="default"
                >
                  foo
                </Section>
              </div>
              <div
                className="dialog-footer"
              >
                <Section
                  bgColor="white"
                  xPadding="default"
                  yPadding="default"
                >
                  <div>
                    foo
                  </div>
                </Section>
              </div>
            </div>
          </div>
        </div>
      </FocusTrap>
    </CSSTransition>
  </Portal>
</Fragment>
`;
